src/renderers/shared/stack/reconciler/__tests__/ReactErrorBoundaries-test.js

```diff
@@ -1193,56 +1434,93 @@ describe('ReactErrorBoundaries', () => {
-      'BrokenComponentDidUpdate render',
-      'BrokenComponentDidUpdate componentDidUpdate [!]',
-      // FIXME: uncomment when componentDidUpdate() gets caught.
-      // Catch and render an error message
-      // 'ErrorBoundary unstable_handleError',
-      // 'ErrorBoundary render error',
-      // 'ErrorBoundary componentDidUpdate',
-    ]);
 
-    log.length = 0;
-    ReactDOM.unmountComponentAtNode(container);
-    expect(log).toEqual([
-      'ErrorBoundary componentWillUnmount',
-      'BrokenComponentDidUpdate componentWillUnmount',
-    ]);
-  });
+      log.length = 0;
+      ReactDOM.render(
+        <ErrorBoundary>
+          <BrokenComponentDidUpdate />
+        </ErrorBoundary>,
+        container
+      );
+      expect(log).toEqual([
+        'ErrorBoundary componentWillReceiveProps',
+        'ErrorBoundary componentWillUpdate',
+        'ErrorBoundary render success',
+        'BrokenComponentDidUpdate componentWillReceiveProps',
+        'BrokenComponentDidUpdate componentWillUpdate',
+        'BrokenComponentDidUpdate render',
+        'BrokenComponentDidUpdate componentDidUpdate [!]',
+        'ErrorBoundary unstable_handleError',
+        'ErrorBoundary componentWillUpdate',
+        'ErrorBoundary render error',
+        'BrokenComponentDidUpdate componentWillUnmount',
+        'ErrorBoundary componentDidUpdate',
+      ]);
+
+      log.length = 0;
+      ReactDOM.unmountComponentAtNode(container);
+      expect(log).toEqual([
+        'ErrorBoundary componentWillUnmount',
+      ]);
+    });
+  } else {
+    // Known limitation because componentDidUpdate() does not occur on the stack.
+    // This is solved in Fiber.
+    it('currently does not catch errors in componentDidUpdate', () => {
+      var container = document.createElement('div');
+      ReactDOM.render(
+        <ErrorBoundary>
+          <BrokenComponentDidUpdate />
+        </ErrorBoundary>,
+        container
+      );
 
-  it('recovers from componentWillUnmount errors on update', () => {
+      log.length = 0;
+      expect(() => {
+        ReactDOM.render(
+          <ErrorBoundary>
+            <BrokenComponentDidUpdate />
+          </ErrorBoundary>,
+          container
+        );
+      }).toThrow();
+      expect(log).toEqual([
+        'ErrorBoundary componentWillReceiveProps',
+        'ErrorBoundary componentWillUpdate',
+        'ErrorBoundary render success',
+        'BrokenComponentDidUpdate componentWillReceiveProps',
+        'BrokenComponentDidUpdate componentWillUpdate',
+        'BrokenComponentDidUpdate render',
+        'BrokenComponentDidUpdate componentDidUpdate [!]',
+        // The error doesn't get caught :-(
+      ]);
+
+      log.length = 0;
+      ReactDOM.unmountComponentAtNode(container);
+      expect(log).toEqual([
+        'ErrorBoundary componentWillUnmount',
+        'BrokenComponentDidUpdate componentWillUnmount',
+      ]);
+    });
+  }
+
+  fit('recovers from componentWillUnmount errors on update', () => {
```

Comment:
It's all broken now :-) I'm just pushing to make myself feel better. I'll push the final version when it's ready.